### DISCUSSION TITLE
[Feature] Different upload location for different users

### DISCUSSION THREAD

**Discussion by Digital39999** [immich-app/immich#6036] [Feature Request]
### The feature

If we have multiple users, or if we would want to change where Immich stores newly uploaded media (ei. if I have all my old images on external drive, I'd like my newly made ones to also be directly just stored on drive instead of server drive and then move them every so few), we could just specify it from admin page, for each user (and default would just use env).

Something like:
![image](https://github.com/immich-app/immich/assets/65065580/3d82c2e4-11d6-4d9e-a8d6-422301a6b4ea)
or
![image](https://github.com/immich-app/immich/assets/65065580/63ebfa41-7fe0-4be0-ad7c-2b3fcbdbea52)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by gabtrek** [comment]
I agree, but I would add the flexibility of choosing a path for Immich folders (encoded-video, profile, thumbs, upload), and a different path for each user in "library", or at least a different path for "library" all together. This way, it's easy for a user to setup a backup method that doesn't copy every thumbnail and encoded video!

**Reply by aviv926** [reply]
Immich has since made some changes to the variables provided in the answer, but I'm sure you'll find it useful:
https://github.com/immich-app/immich/discussions/2110#discussioncomment-5477767

**Comment by aviv926** [comment]
I would love to see an "Add from external library" option.
This option will allow the user to mark some assets that are from an external library and allow the user to transfer them to the internal storage so that there is no dependency on an external library and then the file can be edited.

**Comment by bo0tzz** [comment]
#5984
